### PR TITLE
Private-events app

### PR DESCRIPTION
**Pull request by mugberto**
# Project: Private-event:
This project consist of building a Rails app that allows the user to create event and invite each other.

### CODE REVIEW

File: app/views/layouts/application.html.erb [hunk 1]
```diff
@@ -10,6 +10,19 @@
...
+      <%= link_to "All Events", events_path %> |
+      <%= link_to "Create Event", new_event_path %> | 
+      <%= link_to "Invite a user", new_attendance_path %> | 
+      <%= link_to "log out", user_session_path(current_user.id), method: :delete %> </div>
+    <% else %>
+      <div><%= link_to "Log in", new_user_session_path %> | <%= link_to "Sign up", new_user_path %></div>
+    <% end %>
+    <% flash.each do |name, message| %>
+      <div><%= message %></div>
+    <% end %>
```
Comment (MahmoudBakr23) [inline app/views/layouts/application.html.erb]: Move the logic here into the helpers.

File: app/models/event.rb [hunk 2]
```diff
@@ -0,0 +1,8 @@
+class Event < ApplicationRecord
+  belongs_to :user, class_name: 'User'
```
Comment (MahmoudBakr23) [inline app/models/event.rb]: ```belongs_to :creator, class_name: 'User'```

**Review by MahmoudBakr23 — CHANGES_REQUESTED** [review]
# **Changes Required 🛑♻**

<hr>

Hello there :)
Great work and effort you put into this project. I liked how you managed to invite a user to attend an event. Also, your Linter is configured correctly, short and descriptive commit messages, PR summary, set up routes, controllers, and RSpec tests. Well done! 👍🎉 However, there's always room for improvement. You still have a few more changes to make in order to get the approval ✔👍

<hr>

- [ ] In README file, please add the CLI ```rails db:create``` before ```rails db:migrate``` and ```rails webpacker:install``` at the end to those who are not having webpack installed because you're using webpack
- [ ] Please keep in mind that the purpose of using scopes inside the Event model is to avoid using Enumerable (each method) to separate the "Upcoming events" from the "Past events". In your events controller's index method, kindly add two more instance variables ```@upcoming_events``` and ```@past_events``` using the scopes you already made, then refactor the related view file to fit this logic.
- [ ] Kindly move the logic from the application view file into the helpers. See my comments on the code!
- [ ] I like the invitation feature, but please make it possible for a user to attend an event on his own, not only by invites.
- [ ] Please make sure to follow the project specifications about the FKs. In your Event model, it should be: ```belongs_to :creator, class_name: 'User'``` and please keep in mind that class_names are used to make rails understand that whatever the previous name is, make sure to refer to the original class_name (Model) which in our case the User.
user_assoc_requirement

- [ ] Following the previous point: Please keep in mind that after you modify the Event model you'll have to modify your migrations so that it reflects correctly the foreign keys and indexes in your schema
foreign_key_index_requirement
Hint:
```
def change
  create_table :events do |t|
    t.integer :creator_id
  end

  add_index :events, :creator_id
end
```
<hr>

## **[OPTIONAL]**

- The user who created an event becomes the owner, so it doesn't make any sense if he's not attending the event by default!
- Your extra feature is cool, but it has some flaws. First: only the user that created the event can invite other users. Please fix that issue only to make more sense.
- Following: A user can't invite himself to an event. Especially if he's not the owner of this event! Kindly add the proper logic to fix this issue as well

<hr>

Kindly make the required changes and submit for another CRR.
Please feel free to contact me on Slack with the name "Mahmoud Bakr" if you had any questions regarding my review! 👍 
Good luck and happy coding 💻👨‍💻

**Review by MahmoudBakr23 — CHANGES_REQUESTED** [review]
# **Changes Required 🛑♻**

<hr>

Hello again :)
You did great making the required changes. Well done! However, there is one point left that needs to be implemented, and I'm sure it'll be done in no time 💪👍

<hr>

- [ ] Please refer to the first review and make sure to make the last two required changes. The first one is to make the event belongs to the creator (User), and the second is to adjust the migrations to reflect that change in the schema.

<hr>

Good luck and happy coding 👨‍💻💻

**Review by rindrajosia — CHANGES_REQUESTED** [review]
Changes Required :red_circle:
To Highlight: :angel:

- [ ] It has functional login page :+1: 
- [ ]  Logged in user can create new event :100: 
- [ ]  the events table has column creator_id of type integer :1st_place_medal: 
- [ ] an index for creator_id column is added :100: 

To Improve :factory_worker:

- [ ]  In your UsersController#show action you need also:

1.  lists events previoulsy attendented by user **using event model scope**, i.e. @past_events = current_user.attended_events.previous ([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/rails/associations/user_events_requirement.png) from [Odin](https://www.theodinproject.com/courses/ruby-on-rails/lessons/associations))
2. lists upcoming events previously attended by user **using event model scope**, i.e. @upcoming_events = current_user.attended_events.upcoming ([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/rails/associations/user_events_requirement.png) from [Odin](https://www.theodinproject.com/courses/ruby-on-rails/lessons/associations)).

- [ ] I suggest that you add screenshot In your readme file.
- [ ] File '.stylelintrc.json' and  '.rubocop.yml' are missing in your project, make sure that they are present because they help you to improve your codebase quality


Nice to have

- [ ]  It will be nice if the event's owner attend the event by default.

**Comment by mugberto** [comment]
- I am using  Linters's config files '.stylelintrc.json' and '.rubocop.yml' locally. They are gitignored.
- I appreciate the suggestion but I will add a screenshot in the readme once I finish making my UI pretty which is not a requirement for this project. I am in a hurry because I am on an improvement period! I need to submit projects

**Review by andresporras3423 — CHANGES_REQUESTED** [review]
### Status: Changes Required :recycle:
**Hello Hubert Mugabo** 
Good job with the improvements for this project :muscle:  :muscle:
It meets most of the requirements :clap: :clap: :clap:  
Just a couple of details to improve before you can go ahead:

- [ ] Fix rubocop offenses.
- [ ] In your README file, "Setup configurations" the name of the repo to clone is wrong.
 
Please request another review once you are ready! :+1: 
 
Happy coding :relaxed:
 
Oscar Andrés Russi Porras

**Review by smcommits — APPROVED** [review]
# Status: Approved :+1: 
## Hello, @mugberto! You have done a great job. The application appears to be amazing. Nice work addressing all the issues that were pointed out by the previous TSEs. This PR is now approved. You can go ahead and merge the branch. Congratulations and good luck with your next project.

**Pull request merged**